src/librustc_lint/lib.rs

```diff
@@ -497,7 +497,7 @@ pub fn register_internals(store: &mut lint::LintStore, sess: Option<&Session>) {
     store.register_group(
         sess,
         false,
-        "internal",
+        "rustc::internal",
```

Comment:
Is this how are we doing that in clippy?

In any case, you'll also need to add it to https://github.com/rust-lang/rust/blob/1eaaf440d5173f090d6e937f4b4cffec6c038984/src/libsyntax/attr/mod.rs#L63

Reply:
Lint groups aren't really tool lints, since they're just registered by a plain string and not with a static variable, like normal lints.

In Clippy we're doing this also with just prepending `clippy::` to the group:
https://github.com/rust-lang/rust-clippy/blob/868f168c1c5fa9f19c922f3b0ba5be3980f6b0a9/clippy_lints/src/lib.rs#L586